### DISCUSSION TITLE
Example projects list on the model of extensions list

### DISCUSSION THREAD

**Discussion by Bouh** [4ian/GDevelop#2467] [Enhancement ideas (and their technical discussions)]
# Current

The example projects are bundled with the app. There is more and more example over time. Lot of examples are submit on the GDevelop repo by different users.
There is feature example, or game example.

# Expected
Move all game example on a list, based on the model of extension list.
I like the way to just download what we want.
We could have a list of examples downloadable. This way we can add more examples or template for different types of games.
Approve more and more of their example without bloat the app and avoid to waiting for the next release.

The existing examples are feature example and can be keeped in the current list, a check of example can be required for see if the example is uptodate, useless, wrong, not working anymore, or just has to be moved with game examples.

**Comment by Bouh** [comment]
All project examples start to take a lot of space.
When GDevelop use 389Mo on the disk, the example use 61Mo.
Reduce the data from the bundle is also a way to reduce the data and time of updates from the auto-updater.

Like the extensions are pulled from Github, the examples project can use the same system.
![image](https://user-images.githubusercontent.com/1670670/118014056-158aaf80-b353-11eb-9a14-c698c8208f49.png)

**Comment by Midhil457** [comment]
One question, if an example made in the latest version is loaded on an older version (Which might not have all the actions used in the example) won't it break?

**Reply by 4ian** [reply]
Yep it would break/not work. That's why we would probably add a "minimumGDevelopVersion" field to the example, or just only allow examples to be opened if they were done on a version lower than the current one (and ask to upgrade otherwise).

**Reply by tristanbob** [reply]
Very good question.  How do extensions work when someone is on a version lower than the extension is built on?

- Does it not show the extension in the catalog? 
- Show the extension but give error if they try to install it?
- No safety checks currently implemented?

**Reply by 4ian** [reply]
By default extensions are considered compatible with whatever version of GDevelop you're using (though this is not the case in practice when using new features, but GDevelop new version adoption is almost 100%).

If there is a known incompatibility, we can set in the extension (manually, in the JSON file) the field `gdevelopVersion`. Any version older than this one will then display 
"Unfortunately, this extension requires a newer version of GDevelop to work. Upgrade GDevelop to be able to use this extension in your project."
and the button to install it will be greyed.

**Comment by Bouh** [comment]
Close by  #2697
